### PR TITLE
[C-2042] Fix empty collection on deep link

### PR DESCRIPTION
**Pull request by sliptype**
### Description

* Fix empty collection lineup upon deep linking due not waiting for the reset action to complete before triggering the fetch
* Sequence reset and fetch actions by putting into a new saga

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/web/src/common/store/pages/collection/sagas.js [hunk 1]
```diff
@@ -76,6 +86,42 @@ function* watchResetCollection() {
...
+      }
+
+      // Need to refetch the collection after resetting
+      // Will pull from cache if it exists
+      // TODO: fix this for smart collections
+      if (typeof collectionId === 'number') {
+        yield put(fetchCollection(action.collectionId))
+      }
+
+      yield take(collectionActions.fetchCollectionSucceeded)
```
Comment (amendelsohn) [inline packages/web/src/common/store/pages/collection/sagas.js:113]: Nice! Good benefit of switching to a saga

File: packages/mobile/src/screens/collection-screen/CollectionScreenDetailsTile.tsx [hunk 2]
```diff
@@ -89,20 +87,9 @@ export const CollectionScreenDetailsTile = ({
...
-
-    // Need to refetch the collection after resetting
-    // Will pull from cache if it exists
-    // TODO: fix this for smart collections
-    if (typeof collectionId === 'number') {
-      dispatch(fetchCollection(collectionId))
-    }
-    dispatch(tracksActions.fetchLineupMetadatas(0, 200, false, undefined))
-  }, [dispatch, collectionUid, userUid, collectionId])
+    dispatch(resetAndFetchCollection(collectionId))
```
Comment (amendelsohn) [inline packages/mobile/src/screens/collection-screen/CollectionScreenDetailsTile.tsx:91]: this doesn't match the name in the actions file `resetAndFetchCollectionTracks`

File: packages/web/src/common/store/pages/collection/sagas.js [hunk 3]
```diff
@@ -76,6 +86,42 @@ function* watchResetCollection() {
...
+    collectionActions.RESET_AND_FETCH_COLLECTION_TRACKS,
+    function* (action) {
+      const collectionUid = yield select(getCollectionUid)
+      const userUid = yield select(getUserUid)
+
+      if (collectionUid && userUid) {
+        // Reset collection so that lineup is not shared between separate instances
+        // of the CollectionScreen
+        yield put(collectionActions.resetCollection(collectionUid, userUid))
+        yield take(tracksActions.reset().type)
```
Comment (amendelsohn) [inline packages/web/src/common/store/pages/collection/sagas.js:103]: why do we have to do this one differently than the others like `collectionActions.fetchCollectionSucceeded`

**Review by amendelsohn — APPROVED** [review]
Very nice! Feels better to have the timing more explicitly controlled.

File: packages/mobile/src/screens/collection-screen/CollectionScreenDetailsTile.tsx (hunk 2 above)
Comment (sliptype) [reply packages/mobile/src/screens/collection-screen/CollectionScreenDetailsTile.tsx]: Updated that right at the end, fixed!

File: packages/web/src/common/store/pages/collection/sagas.js (hunk 3 above)
Comment (sliptype) [reply packages/web/src/common/store/pages/collection/sagas.js:103]: This stumped me for a bit. I guess by default `take` calls the function if it's an action creator and uses the `type` value from the returned action. Well our lineup action creators use the `this` context to add a prefix, which was undefined when called by `take` implicitly

https://github.com/AudiusProject/audius-client/blob/main/packages/common/src/store/lineup/actions.ts#L208

Comment (amendelsohn) [reply packages/web/src/common/store/pages/collection/sagas.js:103]: AHHHH cool. I remember seeing that on the lineup actions. Good to know!

**Pull request merged**